src/CommandingOfficers/Tech.java

```diff
@@ -0,0 +1,514 @@
+    if( getActiveAbilityName().contentEquals(OVERCHARGE_NAME))
+    for(Unit u: units)
+    {
+      if( u.getPreciseHP() > u.model.maxHP ) u.alterHP(10);
+    }
+
+    // Do the normal turn-init stuff.
+    GameEventQueue events = super.initTurn(map);
+
+    return events;
+  }
+
+  /** Heal all units by the specified amount, allowing HP>10, and provide a buff. */
+  private static class OverchargeAbility extends CommanderAbility
+  {
+    private CODamageModifier damageBuff = null;
+    private CODefenseModifier defenseBuff = null;
+    private int healAmount;
+
+    public OverchargeAbility(Commander commander, String abilityName, double abilityCost, int buff, int healAmt)
+    {
+      super(commander, abilityName, abilityCost);
+
+      // Create COModifiers that we can apply when needed.
+      damageBuff = new CODamageModifier(buff);
+      defenseBuff = new CODefenseModifier(buff);
+      healAmount = healAmt;
+
+      // Only mechanical units get the firepower boost.
+      long exclude = UnitModel.TROOP;
+      ArrayList<UnitModel> models = commander.getAllModels(Tech.mechanicalUnits, true, exclude);
+      for( UnitModel m : models )
+        damageBuff.addApplicableUnitModel(m);
+
+      AIFlags = PHASE_TURN_START | PHASE_TURN_END;
+    }
+
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    protected void perform(MapMaster gameMap)
+    {
+      // Bump up our power level.
+      myCommander.addCOModifier(damageBuff);
+      myCommander.addCOModifier(defenseBuff);
+
+      // Overcharge
+      for(Unit u: myCommander.units)
+      {
+        // Only mechanical units get overcharged.
+        if( u.model.isNone(UnitModel.TROOP) )
+        u.alterHP(healAmount, true);
+      }
+    }
+  }
+
+  /** Drop a MechWarrior onto the most contested point on the battlefront. */
+  private static class TechdropAbility extends CommanderAbility
+  {
+    private static final long serialVersionUID = 1L;
+
+    private CODamageModifier damageBuff = null;
+    private CODefenseModifier defenseBuff = null;
+    private int dropRange;
+    private int numDrops;
+    private UnitModel unitModelToDrop;
+
+    GameEventQueue abilityEvents;
+
+    TechdropAbility(Commander commander, String abilityName, double abilityCost, UnitModel unitToDrop, int buff, int num, int abilityRange)
+    {
+      super(commander, abilityName, abilityCost);
+
+      unitModelToDrop = unitToDrop;
+
+      // Create COModifiers that we can apply when needed.
+      damageBuff = new CODamageModifier(buff);
+      defenseBuff = new CODefenseModifier(buff);
+      numDrops = num;
+      dropRange = abilityRange;
+      abilityEvents = new GameEventQueue();
+
+      // Only mechanical units get the firepower boost.
+      long exclude = UnitModel.TROOP;
+      ArrayList<UnitModel> models = commander.getAllModels(Tech.mechanicalUnits, true, exclude);
+      for( UnitModel m : models )
+        damageBuff.addApplicableUnitModel(m);
```

Comment:
Why not just have Tech do the filtering once in her constructor, and pass in a strongly-typed `Tech` instead of `Commander`?

Reply:
Guess I could. Make it static even, so no need to cast. It didn't occur to me at the time, and it's not an expensive or frequent operation anyway.

Reply:
Can't be static, because each `Tech` has different models.

Reply:
Right.